mirrord/cli/src/connection.rs

```diff
@@ -81,8 +82,21 @@ where
     let api = api.prepare_client_cert(analytics).await.into_certified()?;
     user_cert_subtask.success(Some("user credentials prepared"));
 
+    // TODO(alex) [high] 3: I don't think we can `ResolveTarget` here, it must be done only
+    // after we decide if it should be copy or not? Or at least, if we can't find a valid
+    // target, then it should become a copy?
```

Comment:
Then why is it here, I don't get it 😬

Reply:
To avoid this kind of confusion, I was thinking of _statefying_ the `ResolvedTarget`. Something along the lines of `ResolvedTarget<Unchecked>` and `ResolvedTarget<Checked>`, wdyt?

Reply:
Maybe something like this [moving states into a type parameter](https://willcrichton.net/rust-api-type-patterns/typestate.html#moving-states-into-a-type-parameter)

Reply:
Not sure if it's necessary, but I'm fine with that, I think.
And BTW, is state as generic type parameter better than state as a const generics bool?

Reply:
Considering that it's just 2 states, const generics bool is way easier to implement for an enum!